lib/spreewald/web_steps.rb

```diff
@@ -217,14 +217,23 @@
 #
 # See [here](https://makandracards.com/makandra/1225-test-that-a-number-or-money-amount-is-shown-with-cucumber) for details
 Then /^I should( not)? see the (?:number|amount) ([\-\d,\.]+)(?: (.*?))?$/ do |negate, amount, unit|
-  no_minus = amount.starts_with?('-') ? '' : '[^\\-]'
-  nbsp = " "
-  regexp = Regexp.new(no_minus + "\\b" + Regexp.quote(amount) + (unit ? "( |#{nbsp}|&nbsp;)(#{unit}|#{Regexp.quote(HTMLEntities.new.encode(unit, :named))})" :"\\b"))
-  expectation = negate ? :not_to : :to
+  expect_to_match = negate ? :not_to : :to
+  amount_included_minus = amount.gsub!(/^-/, '')
```

Comment:
I had to think a bit to understand this line - I think it's a little bit too clever. There are two things happening: 
1.) Check if amount starts with a `-` and 
2.) Remove that. 
Also the variable name can be read two ways "amount with minus" or "formerly amount included a minus" (I understand you meant the second one).
Please rename `amount_included_minus` to something like `is_negative` to make things clearer. 
If you prefer to keep the check and removal in one line I don't like it I'm okay with it.